stock_request/models/stock_request_order.py

```diff
@@ -139,10 +139,78 @@ def _get_default_requested_by(self):
         string="Stock requests", compute="_compute_stock_request_count", readonly=True
     )
 
+    route_ids = fields.Many2many(
+        "stock.location.route",
+        string="Routes",
+        compute="_compute_route_ids",
+        readonly=True,
+    )
+
+    route_id = fields.Many2one(
+        "stock.location.route",
+        compute="_compute_route_id",
+        inverse="_inverse_route_id",
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        store=True,
+        help="The route related to a stock request order",
+        domain="[('id', 'in', route_ids)]",
+    )
+
     _sql_constraints = [
         ("name_uniq", "unique(name, company_id)", "Stock Request name must be unique")
     ]
 
+    @api.depends("warehouse_id", "location_id")
+    def _compute_route_ids(self):
+        route_obj = self.env["stock.location.route"]
+        routes = route_obj.search(
+            [("warehouse_ids", "in", self.mapped("warehouse_id").ids)]
+        )
+        routes_by_warehouse = {}
+        for route in routes:
+            for warehouse in route.warehouse_ids:
+                routes_by_warehouse.setdefault(
+                    warehouse.id, self.env["stock.location.route"]
+                )
+                routes_by_warehouse[warehouse.id] |= route
+        for record in self:
+            routes = route_obj
+            if record.warehouse_id and routes_by_warehouse.get(record.warehouse_id.id):
+                routes |= routes_by_warehouse[record.warehouse_id.id]
+            parents = record.get_parents().ids
+            record.route_ids = routes.filtered(
+                lambda r: any(p.location_id.id in parents for p in r.rule_ids)
+            )
```

Comment:
This one has a couple of problems:

1. It is duplicating code with the equivalent method of stock.request and making maintenance a bit more difficutl.
2. It is completely ignoring product and product category routes making it have less fewer possibilities than what actually would have been correct to propose.

Proposed approach: just combine the route_ids of all lines.

Downside: if a route is feasible for one line and not for another line one of the requests might not work. To avoid problems, I would add a constraint on stock.request to check that the relected route_id is among route_ids at the time of validation.